websocket/src/test/java/org/nanohttpd/junit/protocols/websockets/WebSocketFrameTest.java

```diff
@@ -0,0 +1,140 @@
+import java.util.ArrayList;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.nanohttpd.protocols.websockets.OpCode;
+import org.nanohttpd.protocols.websockets.WebSocketException;
+import org.nanohttpd.protocols.websockets.WebSocketFrame;
+
+public class WebSocketFrameTest {
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void testReadEOFException() throws IOException {
+
+        // Arrange
+        final byte[] byteArray = {0};
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray);
+
+        // Act
+        thrown.expect(EOFException.class);
+        WebSocketFrame.read(in);
+
+        // Method is not expected to return due to exception thrown
+    }
+
+    @Test
+    public void testReadWebSocketExceptionMaxFrameLength() throws IOException {
+
+        // Arrange
+        final byte[] byteArray = {-128, -1, 0, 0, 0, 1, 47, 120, -16, 16};
+
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        WebSocketFrame.read(in);
+    }
+
+    @Test
+    public void testReadWebSocketException4ByteLength() throws Exception {
+
+        // Arrange
+        final byte[] byteArray = {-128, -1, -128, 0, 0, 0, 47, 120, -16, 16
+        };
+
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        WebSocketFrame.read(in);
+    }
+
+    @Test
+    public void testReadWebSocketException2ByteLength() throws Exception {
+
+        // Arrange
+        final byte[] byteArray = {-59, -128, 126, 0, 0, 32, -1, -128, 4, -128};
+
+        // Arrange
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray, 1, 9);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        WebSocketFrame.read(in);
+
+    }
+
+    @Test
+    public void testReadPayloadInfoReservedBitsNotZero() throws Throwable {
+
+        // Arrange
+        final byte[] byteArray = {24, 127, 0, 0, 0, 0, 0, -128, 0, 1};
+        final ByteArrayInputStream in = new ByteArrayInputStream(byteArray);
+        final ArrayList<WebSocketFrame> fragments = new ArrayList<WebSocketFrame>();
+        final WebSocketFrame objectUnderTest = new WebSocketFrame(OpCode.Close, fragments);
+
+        // Act
+        thrown.expect(WebSocketException.class);
+        objectUnderTest.read(in);
+
+    }
+
+    @Test
+    public void testSetTextPayload() throws WebSocketException, CharacterCodingException {
+
+        // Arrange
+        final ArrayList<WebSocketFrame> fragments = new ArrayList<WebSocketFrame>();
+        final String payload = "foo";
+
+        final WebSocketFrame objectUnderTest = new WebSocketFrame(OpCode.Text, fragments);
+
+        // Act
+        objectUnderTest.setTextPayload(payload);
+
+        // Assert side effects
```

Comment:
Might it be confusing to call this a "side effect", given that this assert tests the only intended effect of the method?

Reply:
"side effect" means observable behaviour besides the return value (as opposed to a _pure_ function whose only effect is returning a value).

Reply:
OK I understand, I've now removed the `Assert`so I'm not sure how useful the test is.

Reply:
Looks like I misunderstood the comments. I've added the Assert back in but commented it 
`// Assert Act`